src/lib/y2storage/boot_requirements_strategies/analyzer.rb

```diff
@@ -468,6 +488,46 @@ def in_lvm?(device)
         end
       end
 
+      # Whether the device is in a thinly provisioned LVM logical volume
+      #
+      # The device can be a planned one or a filesystem from the devicegraph.
+      #
+      # @param device [Filesystems::Base, Planned::Device, nil]
+      # @return [Boolean] false if device is nil
+      def in_thin_lvm?(device)
+        return false if device.nil?
+
+        if device.is_a?(Planned::Device)
+          device.is_a?(Planned::LvmLv) && device.lv_type == LvType::THIN
+        else
+          # If this is not a BlkFilesystem (e.g. NFS), it can't be on thin LVM
+          return false unless device.respond_to?(:plain_blk_devices)
```

Comment:
NP: why not this? `return false unless device.is?(:blk_filesystem)`

Reply:
Because Ruby fans insist on "duck typing"; just because right now we only have that method for BlkFilesystem that doesn't necessarily mean that one day we won't extend the concept for other classes. That `unless` clause really only defends against a _no method_ error. If the method is there, by all means let's use it. And that's something different than explicitly checking a specific class.

I realize that this is not exactly what the comment in the code says; that comment was meant more as a practical example just _when_ the check is necessary. I myself learned that the hard way when the NFS test cases started failing. ;-)